docs/Migrating_LLB_to_a_PPL_based_Custom_Device.md

```diff
@@ -0,0 +1,74 @@
+* Even if two given custom devices would reference the same file (VI) dependency, by packaging them into PPLs, each custom device would receive its own copy of the required dependency. This would be compiled and included in its respective PPL via name spacing for the contained items, the PPLs would also make sure that load-time conflicts are avoided.
+
+* Since PPLs preserve the file hierarchy of the source project library, that solves the issue of internal naming conflicts, which is specific to LLBs (which causes compile-time warnings and extra files to be generated). Multiple LabVIEW project libraries can be included in the same PPL, even if those contain items with the same name. This is the same reason why LabVIEW classes could now be easier used when building VeriStand Custom Devices.
+
+* Much smaller disk footprint, which is important for a lot of our customers, since it would have a large impact on the deployment time (as you can see for the FPGA Addon below). 
+
+![](images/Disk_Footprint.jpg)
+### Drawbacks of using Packed Project Libraries with VeriStand Custom Devices
+
+* Probably one of the main issues is that inside the PPL, relative paths for the items are maintained starting from the most common directory on the disk. This can be confusing for LabVIEW developers that would expect the relative paths to start from the root of the library and follow the original project structure. Additionally, it also means that changing the organization, or location on the disk, of the items that are included in the library build specification could in turn affect the relative paths of the items in the packed library. In time, this behavior could prove to be quite a challenge (or at least a nuisance), since changes within the custom device folder structure could affect the organization of the items in the generated packed library. In turn, this would create the need to constantly update the configuration XML file with the new paths. The long-term solution to this issue would be to create a post-build step that could perform these operations automatically: i.e. parse the configuration XML file and identify the items with "path" properties. For each item, find the corresponding element within the generated packed library and update the configuration file. Once the update is complete (and with no errors), copy the resulting files to the VeriStand Custom Device directory. Certain LabVIEW features are not compatible with packed project libraries. For instance, malleable VIs (VIMs) cannot be included in a PPL and exported as public members, they can only be set with private scope. Implicitly, if a custom device code makes use of malleable VIs, packing it inside a PPL would require some changes (to either the code or the project level) to maintain compatibility.
```

Comment:
```suggestion
* One of the main drawback is that inside the PPL, relative paths for the items are maintained starting from the most common directory on disk. This can be confusing for LabVIEW developers that would expect the relative paths to start from the root of the library and follow the original project structure. Additionally, it also means that changing the organization, or location on disk, of the items that are included in the library build specification could in turn affect the relative paths of the items in the packed library. In time, this behavior could prove to be quite a challenge (or at least a nuisance), since changes within the custom device folder structure could affect the organization of the items in the generated packed library. In turn, this would create the need to constantly update the configuration XML file with the new paths. The long-term solution to this issue would be to create a post-build step that could perform these operations automatically: i.e. parse the configuration XML file and identify the items with "path" properties. For each item, find the corresponding element within the generated packed library and update the configuration file. Once the update is complete (and with no errors), copy the resulting files to the VeriStand Custom Device directory.
* Certain LabVIEW features are not compatible with packed project libraries. For instance, malleable VIs (VIMs) cannot be included in a PPL and exported as public members, they can only be set with private scope. Implicitly, if a custom device code makes use of malleable VIs, packing it inside a PPL would require some changes (to either the code or the project level) to maintain compatibility.
```